packages/fuels-abigen-macro/tests/harness.rs

```diff
@@ -608,10 +608,10 @@ async fn create_nested_struct_from_decoded_tokens() {
     assert_eq!(10, nested_struct_from_tokens.x);
     assert!(nested_struct_from_tokens.y.a);
 
-    let (provider, wallet) = setup_test_provider_and_wallet().await;
+    let (_, wallet) = setup_test_provider_and_wallet().await;
```

Comment:
the fact that the provider is consistently put as an ignored variable (`_`) tells me that either:
- the function  should be refactored
- a new function should be written

Reply:
I don't have a good enough idea of all the use cases users might have. There are some scenarios where direct interaction with the provider is needed, but it feels awkward do it via the wallet.

Reply:
I think having a function that is just `setup_test_wallet` (that the `setup_test_provider_and_wallet` would use) would be good here! This way you only get what you need in the cases where you used `_`, but we can still get the `provider` when it's needed. What do you think?